src/common/Header/Header.js

```diff
@@ -1,16 +1,39 @@
-import React from "react";
+import React, { useState } from "react";
 import { Logo, Menu } from "./HeaderComponents";
-import "./Header.css";
+import "./Header.scss";
 
 function Header({ image, alt }) {
+    const [opened, setOpened] = useState(false);
+
+    const handleToggle = () => {
+        setOpened(!opened);
+    };
+
+    const styles = {
+        bar1: {
+            transform: opened && "rotate(-45deg) translate(-9px, 6px)"
+        },
+        bar2: {
+            opacity: opened && 0
+        },
+        bar3: {
+            transform: opened && "rotate(45deg) translate(-8px, -8px)"
+        }
+    };
+
     return(
-        <header className="App-header">
-            <div className="App-header-wrapper">
+        <header className="App__header">
+            <div className="App__header__wrapper">
                 <Logo
                     image={image}
                     alt={alt}
                 />
-                <Menu />
+                <Menu opened={opened} />
+                <div className="App__header__hamburger" onClick={handleToggle}>
+                    <div className="App__header__hamburger__bar1" style={styles.bar1} ></div>
+                    <div className="App__header__hamburger__bar2" style={styles.bar2}></div>
+                    <div className="App__header__hamburger__bar3" style={styles.bar3}></div>
+                </div>
             </div>
         </header>

```

Comment:
Please consider more simple notation e.g. `header` instead of `App__header` and forward elements can be marked as `header__hamburger`, `header__bar1`, `header__bar2`, etc. because the main container is `header` there is no need to explicitly show nesting while naming classes :)